File: projects/10/ruby/Penguin.rb
# Penguin.rb | Defines the Penguin class which inherits attributes and methods
#   from the WalkingBird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Jacob Brink
# Date: 4/29/2020
# Project: 10
####################################################

require './WalkingBird.rb'

class Penguin < WalkingBird
  def call
    'Whoo-hoo'
  end
end

File: projects/10/java/Penguin.java
/* Penguin.java provides an Penguin subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Jacob Brink
 * Date: 4/29/2020
 * Project: 10
 ******************************************************/

public class Penguin extends WalkingBird {

    /* explicit constructor
     * Receive: name, a String 
     * PostCond: myName == name.
     */
    public Penguin() {
	super();
    }
    
    public Penguin(String name) {
	super(name);
    }


    /* An Penguin's Call
     * Return: an penguin-call ("Whoo-hoo!").
     */
    public String call() {
	return "*Penguin Noises*";
    }

    
} 

